app/models/user.rb

```diff
@@ -0,0 +1,3 @@
+class User < ApplicationRecord
+  validates :username, :email, presence: true
```

Comment:
A question: would username or email need to be unique?

Reply:
Yes! you are absolutely right. I will add these constraints.